Source/WebCore/page/EventHandler.cpp

```diff
@@ -594,7 +620,11 @@ void EventHandler::selectClosestWordFromHitTestResult(const HitTestResult& resul
         if (appendTrailingWhitespace == ShouldAppendTrailingWhitespace && newSelection.isRange())
             newSelection.appendTrailingWhitespace();
 
-        updateSelectionForMouseDownDispatchingSelectStart(targetNode.get(), expandSelectionToRespectSelectOnMouseDown(*targetNode, newSelection), TextGranularity::WordGranularity);
+        newSelection = expandSelectionToRespectSelectOnMouseDown(*targetNode, newSelection);
+        if (shouldAvoidExtendingSelectionOnClick(*targetNode, newSelection))
+            return;
+
+        updateSelectionForMouseDownDispatchingSelectStart(targetNode.get(), newSelection, TextGranularity::WordGranularity);
```

Comment:
Nit: May be able to re-use this snippet for `[Word|Paragraph]Granularity` -- `bool updateSelectionForMouseDownDipsatchingSelectStartIfNeeded(..., TextGranularity);`?

Reply:
Or not, it's only a couple of lines.

Reply:
Seems reasonable to me!